moveos/moveos/src/moveos.rs

```diff
@@ -243,12 +268,75 @@ impl MoveOS {
         }
     }
 
-    fn execute_gas_validate(
-        &self,
-        session: &mut MoveOSSession<'_, '_, MoveOSResolverProxy<MoveOSStore>, MoveOSGasMeter>,
-        action: &VerifiedMoveAction,
-    ) -> VMResult<Option<bool>> {
-        match action {
+    pub fn get_address_balance(&self, tx: &MoveOSTransaction) -> VMResult<u128> {
+        let MoveOSTransaction {
+            ctx,
+            action,
+            pre_execute_functions: _pre_execute_functions,
+            post_execute_functions: _post_execute_functions,
+        } = tx;
+
+        let cost_table = initial_cost_schedule();
+        let mut gas_meter = MoveOSGasMeter::new(cost_table, ctx.max_gas_amount);
+        gas_meter.set_metering(false);
+
+        let mut session = self
+            .vm
+            .new_readonly_session(&self.db, ctx.clone(), gas_meter);
+
+        let verified_action = session.verify_move_action(action.clone())?;
+
+        match verified_action {
+            VerifiedMoveAction::Function { call } => {
+                let module_address = call.function_id.module_id.address();
+
+                let gas_coin_module_id = ModuleId::new(
+                    AccountAddress::from_hex_literal("0x3").unwrap(),
+                    Identifier::from(IdentStr::new("gas_coin").unwrap()),
+                );
+                let gas_validate_func_call = FunctionCall::new(
+                    FunctionId::new(gas_coin_module_id, Identifier::new("balance").unwrap()),
+                    vec![],
+                    vec![MoveValue::Address(*module_address)
+                        .simple_serialize()
+                        .unwrap()],
+                );
+
+                let return_value =
+                    session.execute_function_bypass_visibility(gas_validate_func_call)?;
+
+                let first_return_value = return_value.get(0).unwrap();
+
+                let balance = bcs::from_bytes::<move_core_types::u256::U256>(
+                    first_return_value.value.as_slice(),
+                )
+                .expect("the return value of gas validate function should be u128");
+
+                Ok(balance.unchecked_as_u128())
+            }
+            _ => Ok(0),
+        }
+    }
+
+    pub fn execute_gas_validate(&self, tx: &MoveOSTransaction) -> VMResult<Option<bool>> {
+        let MoveOSTransaction {
```

Comment:
Same as above, we can directly call the validate function like the `transaction_validator::validate` in the executor. https://github.com/rooch-network/rooch/blob/6265cd36d2f19e6c42955d53f115dd2dd5af1c73/crates/rooch-types/src/framework/transaction_validator.rs#L29-L57

Reply:
OK, it's fixed.